### DISCUSSION TITLE
Query current volume of a single speaker in a group

### DISCUSSION THREAD

**Discussion by thkleinert** [B5r1oJ0A9G/teufel_raumfeld#40] [Q&A]
Hi,

thanks for all the work you put into this integration! 

Is it possible to query the volume of a specific speaker which is grouped with others? My use case would be to have a slider in home assistant for each Raumfeld speaker to control its volume. I am aware that I can set absolute volumes for individual speakers also if they are grouped. But to keep the sliders in sync (e.g. in case the volume is changed directly at the speaker) I need to be able to query the current volume. With home assistants media player I can only query the volume of the whole group (which I think defaults to the loudest speaker in the group). 

Is something like this possible?

**Comment by B5r1oJ0A9G** [comment]
Hi!

While ``hassfeld`` supports it via `async_get_room_volume(room)`, to enable in ``teufel_raumfeld`` volume control in Spotify SingleRoom mode, this function is not exposed in non-SingleRoom mode.

**Comment by B5r1oJ0A9G** [comment]
FYI: In the current master branch you'll find a new implementation with number entities for setting the room volume:
![Screenshot from 2023-08-25 19-58-00](https://github.com/B5r1oJ0A9G/teufel_raumfeld/assets/1625487/fde80803-6bd0-4ef8-9247-c99a1174763f)

**Comment by DonaldGermany** [comment]
That Volume Card looks amazing, can you give me a hint how to use it? I'm using latest version available on HACS.

**Reply by DonaldGermany** [reply]
OK got it. You only need to use latest files from github :-)
